src/notebook/notebook/widget.ts

```diff
@@ -716,11 +832,31 @@ class Notebook extends StaticNotebook {
   }
 
   /**
+   * Update the active cell when a cell is added or removed.
+   */
+  private _updateActiveCell(): void {
+    let count = this.childCount();
```

Comment:
This logic to clamp the active cell index and emit a signal is also contained in the `set activeCellIndex` function. It would be great to consolidate the logic into one place.